src/lib/holocene/scroll-to-top.svelte

```diff
@@ -0,0 +1,49 @@
+<script lang="ts">
+  import Button from '$holocene/button.svelte';
+
+  export let showOn = 150; // pixels
+  export let scrollToContainer = false;
+  let hidden = true;
+
+  function getScrollContainer(): HTMLElement | null {
+    return scrollToContainer
+      ? document.getElementById('scroll-container')
+      : document.documentElement || document.body;
+  }
+
+  function scrollToTop() {
+    getScrollContainer()?.scrollIntoView();
+  }
+
+  function handleOnScroll() {
+    const scrollContainer = getScrollContainer();
+
+    if (!scrollContainer) return;
+
+    hidden = Boolean(scrollContainer.getBoundingClientRect().top + showOn > 0);
+  }
+</script>
+
+<svelte:window on:scroll={handleOnScroll} />
+
+<div id="scroll-container" class={$$props.class}>
+  <slot />
+  <div class="back-to-top" class:hidden>
+    <Button
+      class="!py-1.5 !px-1"
```

Comment:
What does the ! in front of the class do? I've never seen that with Tailwind

Reply:
I'm using the [important modifier ](https://tailwindcss.com/docs/configuration#important-modifier) here because there is some variant styling (specifically `padding`) applied as `postcss` on the `button` component which overrides the `classes` being passed down to the component.

The other solution here would be to create a `icon` variant and add specific padding for it in the `postcss` on the `button` component, but I'm not so sure we want the same padding for all icons. Hence this hack 😅